ion.js

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+const CORE_ADDON_ID = "ion-core-addon@mozilla.org";
+const ION_SIGNUP_URL = "https://mozilla-ion.github.io/ion-core-addon/";
+
+class Ion {
+  /**
+   * Initialize the Ion library.
+   */
+  async initialize() {
+    console.debug("Ion.initialize");
+
+    await this._checkIonCore().then(
+        () => console.debug("Ion.initialize - Found Ion Core")
+      ).catch(
+        async () => await browser.tabs.create({ url: ION_SIGNUP_URL })
+      );
+
+    // Listen for incoming messages from the Core addon.
+    browser.runtime.onMessageExternal.addListener(
+      (m, s) => this._handleExternalMessage(m, s));
+  }
+
+  /**
+   * Check if the Core addon is installed.
+   *
+   * @returns {Promise} resolved if the core addon was found and
+   *          communication was successful, rejected otherwise.
+   */
+  async _checkIonCore() {
+    try {
+      return await browser.management.get(CORE_ADDON_ID);
+    } catch (ex) {
+      console.error("Ion._checkIonCore - core addon not found", ex);
+      return Promise.reject(ex);
+    }
+
+    // TODO: in addition to checking if the addon is installed,
+    // this should check if user has joined the platform by sending
+    // a message to the addon and waiting for its response.
```

Comment:
This is rather important, but I think can be tackled separately.

Reply:
This sounds like the right approach to me please file an issue!

Reply:
Lovely, I filed #9